packages/mui-utils/src/getReactElementRef/getReactElementRef.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+/**
+ * Returns the ref of a React element handling differences between React 19 and older versions.
+ * It will throw runtime error if the element is not a valid React element.
+ *
+ * @param element React.ReactElement
+ * @returns React.Ref<any> | null | undefined
+ */
+export default function getReactElementRef(
+  element: React.ReactElement,
+): React.Ref<any> | null | undefined {
+  const reactMajorVersion = parseInt(React.version.split('.')[0], 10);
+
+  // 'ref' is passed as prop in React 19, whereas 'ref' is directly attached to children in older versions
+  if (reactMajorVersion >= 19) {
```

Comment:
We could potential save bunch of characters by using string comparison no?

```suggestion
  if (React.version > '19.0.0') {
```

Reply:
went with parseInt, as js acting weird with string checking if version is greater than 100 😆 .

<img width="437" alt="image" src="https://github.com/user-attachments/assets/eef7f0a7-2656-4000-8556-b23b90a6bafd">

Reply:
> tring checking if version is greater than 100 😆 .

At this point, the logic will be gone, but fair enough. Let's fix this one too: #43820.